jwst/group_scale/group_scale_step.py

```diff
@@ -18,37 +18,40 @@ def process(self, input):
         # Open the input data model
         with datamodels.RampModel(input) as input_model:
 
+            # Always work on a copy
+            result = input_model.copy()
```

Comment:
Plagiarism!

Reply:
Ironically, I had to copy and paste the word "plagiarism" from a dictionary site in order to get the spelling correct